graphql_compiler/compiler/emit_sql.py

```diff
@@ -239,19 +338,182 @@ def start_global_operations(self):
 
     def filter(self, predicate):
         """Execute a Filter Block."""
-        sql_expression = predicate.to_sql(self._aliases, self._current_alias)
+        left_predicate_folded = isinstance(predicate.left, FoldedContextField)
+        right_predicate_folded = isinstance(predicate.right, FoldedContextField)
+        if self._in_fold or left_predicate_folded or right_predicate_folded:
+            raise NotImplementedError('Filters inside a fold are not implemented yet.')
+
+        sql_expression = predicate.to_sql(self._aliases,
+                                          self._current_alias)
         if self._is_in_optional_scope():
             sql_expression = sqlalchemy.or_(sql_expression,
                                             self._came_from[self._current_alias].is_(None))
         self._filters.append(sql_expression)
 
+    def get_join_descriptor(self, fold_scope_location):
+        """Convert a fold scope location to a join descriptor from the schema."""
+        # collect edge information to join the fold subquery to the main selectable
+        edge_direction, edge_name = fold_scope_location.fold_path[0]
+        full_edge_name = '{}_{}'.format(edge_direction, edge_name)
+        return self._sql_schema_info.join_descriptors[self._current_classname][full_edge_name]
+
+    def get_fold_info(self, fold_scope_location):
+        """Get the basic from/join info for tables involved in fold w/ their join descriptors."""
+        # basic info about the outer vertex and inner vertex
+        inner_vertex_alias = self._sql_schema_info.vertex_name_to_table[
+            self._ir.query_metadata_table.get_location_info(fold_scope_location).type.name
+        ].alias()
+        outer_vertex_alias = self._current_alias.alias()
+        join_descriptor = self.get_join_descriptor(fold_scope_location)
+
+        return inner_vertex_alias, outer_vertex_alias, join_descriptor
+
+    def get_fold_outputs(self, fold_scope_location, join_descriptor):
```

Comment:
Seems like this and other fold helper functions are intended for internal use only so should start with `_`